android/src/com/mapswithme/maps/location/LocationHelper.java

```diff
@@ -209,6 +209,11 @@ public void run()
   {
     mLogger.d("ctor()");
 
+    // TODO consider refactoring.
+    // Actually we shouldn't initialize Framework here,
+    // to allow app components to retrieve location updates without all heavy framework's stuff initialized.
+    // For now this is necessary to connect mModeChangeListener below.
+    MwmApplication.get().initNativeCore();
```

Comment:
Yep, now it's required at least to set that listener. Left a TODO to refactor that.